stock_picking_auto_create_lot_qty/models/stock_picking.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (C) 2023 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def _set_auto_lot(self):
```

Comment:
Once again, unless there's a strong reason not to, I think it'd be best to split `move_line_ids` by quantity here and then call super() in order to better integrate with other modules.

Reply:
If this function worked with single record i can accept your suggestion, but it is always work with multiple records, and if i will call super(), we will have undisireable result

Reply:
After some exploration, I realized that the method responsible for splitting a picking into separate move lines is `_action_assign`

https://github.com/OCA/OCB/blob/14.0/addons/stock/models/stock_move.py#L1310

So the splitting should be done there IMO. It should be easier to split the moves you need (only if they're tracked by lot and meet other conditions) and then call super() so as to be less destructive.

Reply:
done

Reply:
@dessanhemrayev @GabbasovDinar what about this?

Reply:
the "_action_assign" method is launched at the very end, and before that the "set_lot_auto" method and its subtypes are triggered, so I think everything is correct here